lib/iris/pandas.py

```diff
@@ -518,88 +517,330 @@ def _get_base(array):
     if not copy:
         _assert_shared(data, series)
-
     return series
 
 
-def as_data_frame(cube, copy=True):
+def as_data_frame(
+    cube,
+    copy=True,
+    add_aux_coords=False,
+    add_cell_measures=False,
+    add_ancillary_variables=False,
+):
     """
     Convert a 2D cube to a Pandas DataFrame.
 
-    Args:
+    :attr:`~iris.cube.Cube.dim_coords` and :attr:`~iris.cube.Cube.data` are
+    flattened into a long-style :class:`~pandas.DataFrame`.  Other
+    :attr:`~iris.cube.Cube.aux_coords`, :attr:`~iris.cube.Cube.aux_coords` and :attr:`~iris.cube.Cube.attributes`
+    may be optionally added as additional :class:`~pandas.DataFrame` columns.
 
-        * cube - The cube to convert to a Pandas DataFrame.
+    Parameters
+    ----------
+    cube: :class:`~iris.cube.Cube`
+        The :class:`~iris.cube.Cube` to be converted to a :class:`pandas.DataFrame`.
+    copy : bool, default=True
+        Whether the :class:`pandas.DataFrame` is a copy of the the Cube
+        :attr:`~iris.cube.Cube.data`. This option is provided to help with memory
+        size concerns.
+    add_aux_coords : bool, default=False
+        If True, add all :attr:`~iris.cube.Cube.aux_coords` (including scalar
+        coordinates) to the returned :class:`pandas.DataFrame`.
+    add_cell_measures : bool, default=False
+        If True, add :attr:`~iris.cube.Cube.cell_measures` to the returned
+        :class:`pandas.DataFrame`.
+    add_ancillary_variables: bool, default=False
+        If True, add :attr:`~iris.cube.Cube.ancillary_variables` to the returned
+        :class:`pandas.DataFrame`.
 
-    Kwargs:
+    Returns
+    -------
+    :class:`~pandas.DataFrame`
+        A :class:`~pandas.DataFrame` with :class:`~iris.cube.Cube` dimensions
+        forming a :class:`~pandas.MultiIndex`
 
-        * copy - Whether to make a copy of the data.
-                 Defaults to True. Must be True for masked data
-                 and some data types (see notes below).
+    Notes
+    -----
+    Dask ``DataFrame``\\s are not supported.
 
-    .. note::
+    A :class:`~pandas.MultiIndex` :class:`~pandas.DataFrame` is returned by default.
+    Use the :meth:`~pandas.DataFrame.reset_index` to return a
+    :class:`~pandas.DataFrame` without :class:`~pandas.MultiIndex` levels. Use
+    'inplace=True` to preserve memory object reference.
 
-        This function will copy your data by default.
-        If you have a large array that cannot be copied,
-        make sure it is not masked and use copy=False.
+    :class:`~iris.cube.Cube` data `dtype` is preserved.
 
-    .. note::
+    Warnings
+    --------
+    Where the :class:`~iris.cube.Cube` contains masked values, these become
+    :data:`numpy.nan` in the returned :class:`~pandas.DataFrame`.
 
-        Pandas will sometimes make a copy of the array,
-        for example when creating from an int32 array.
-        Iris will detect this and raise an exception if copy=False.
+    Examples
```

Comment:
The remaining doctest failures are caused by a few different problems in this section. Basically the code provided below doesn't produce the expected results. It's quite difficult for me to debug this as I don't know how you produced the outputs originally, so it would be great if you could. 

### To debug

```bash
$ cd <your iris directory>
$ conda create -n iris-dev --file requirements/ci/nox.lock/py310-linux-64.lock
$ conda activate iris-dev
$ export PYTHONPATH=$PYTHONPATH:$PWD/lib
$ cd docs/src
$ make doctest

```

### Blank lines

If you get any failures because doctest is expecting `<BLANKLINE>`: just add in that exact text where it is expected.

Reply:
Turns out that once I'd fixed the obvious typos and `<BLANKLINE>` issues, `black` was still trolling me... `pandas` printing of `DataFrames` included trailing whitespace which doctest was trying to match, but which kept being removed by `black`.   Fixed with `# doctest: +NORMALIZE_WHITESPACE`

Reply:
That's a useful trick to know about!